### ISSUE TITLE
Dev. Slowdown: My Studio is timing out very often and requires me to login every time.

### ISSUE THREAD

**Issue by ShakAttacked** [orientechnologies/orientdb#8038]
### OS: Centos

Studio is timing out very often and requires me to login every time. It's frustrating because it loses all my queries from the input.  The worst behavior is when everything looks fine, you type a query and when you execute, it goes to login page and once logged, the query is lost.  So you have to start all over again.

Regards,
Safwan

**Comment by metametametameta** [comment]
True, this was driving me nuts too. I tried a few things, but finally adding 

-Dnetwork.http.sessionExpireTimeout=30000 

to graph server startup seems to help. I guess the default is 300 (5 mins). The 5 minutes is too low especially when we spend more time in a single text box such as when writing a query (your case) or a Javascript function.

**Comment by metametametameta** [comment]
Sorry, looks like -Dnetwork.http.sessionExpireTimeout=30000 isn't helping either.  Is there any other configurable setting?

**Comment by ShakAttacked** [comment]
I have one browser instance open. and I switch to work in Java and when I come back, it doesn't say it timed out. I spend time writing a complicated query only to lose it when I click on run.  It's incredibly frustrating to do the work twice.  I started refreshing the page before writing the query so I don't lose my query but unfortunately I'm human and forget.

**Comment by ShakAttacked** [comment]
Hi @luigidellaquila 

Are you guys able to reproduce this, I've tried under 2 different browsers. It would really save us time if we can have Studio fixed ASAP.

Thanks,
Safwan

**Comment by maggiolo00** [comment]
Hi @ShakAttacked 

let me check. Do you also shutdown and restart the server?

**Comment by careerscale** [comment]
@ShakAttacked  didnt notice it. we are using snapshot though.

seem this earlier. but looks ok for me. 

sorry for joining in the middle.

**Comment by ShakAttacked** [comment]
Hi @maggiolo00 , @careerscale 

I'm using the snapshot from the 30th.  I've tried in both chrome and firefox.  I have 3 different problems with Studio.

- It times out in less than 15 minutes (maybe 5). hard to tell as I go to code and come back. As I type a query and execute, it takes me to login page.  this is without a restart of the server.
- all my queries are lost on refresh, I get this in chrome and firefox so it's not browser related. 
- it happens when I restart the server. this happens when I work on a plugin and I need to restart to get the plugin changes (I've opened an issue regarding this, if this is a reload without restart).

Thanks,
Safwan
